src/main/resources/js/updateMacro.js

```diff
@@ -0,0 +1,340 @@
+ * @param macroNode to be updated
+ * @param macroParams existing on the macro that is being updated
+ */
+const addTechnologies = (
+  originalBody,
+  newTechnologies,
+  macroNode,
+  macroParams
+) => {
+  AJS.$("body").append(techInput);
+
+  AJS.$("#idea-technology").auiSelect2({
+    multiple: true,
+    ajax: {
+      url: `${AJS.contextPath()}/rest/idea/1/technology`,
+      data: (params) => ({ q: params }),
+      processResults: (data) => ({
+        results: data.map((item) => ({
+          id: item.label.toLowerCase(),
+          text: item.label,
+        })),
+      }),
+      results: (data) => ({
+        results: data.map((item) => ({
+          id: item.label.toLowerCase(),
+          text: item.label,
+        })),
+      }),
+    },
+  });
+  AJS.dialog2("#edit-page-dialog-tech").show();
+
+  AJS.$("#dialog-submit-button").click((e) => {
+    const oldTechArray = originalBody.split(",");
+
+    // Construct the new technology list
+    const newTechText = [...newTechnologies(), ...oldTechArray]
+      //Remove duplicates
+      .reduce((pre, cur) => {
+        if (
+          pre.includes(cur) ||
+          cur.replaceAll(" ", "") ===
+            " Add your technologies ".replaceAll(" ", "")
+        ) {
+          return pre;
+        } else {
+          return [...pre, cur];
+        }
+      }, [])
+      .join(",");
+
+    e.preventDefault();
+
+    AJS.dialog2("#edit-page-dialog-tech").hide();
+
+    updateMacro(macroNode, macroParams, newTechText);
+
+    AJS.$("#edit-page-dialog-tech").remove();
+  });
+};
+
+/**
+ * Handle the switch of dialog based on the category
+ *
+ * @param e event from confluence
+ * @param macroNode node that the event is on
+ */
+const switchByCategory = (e, macroNode) => {
+  const originalBody = macroNode.textContent;
+
+  const getStringsFromInput = () =>
+    AJS.$(".select2-search-choice")
+      .toArray()
+      .map((item) => item.innerText);
+
+  /**
+   * Create the link to a user in confluence
+   *
+   * @returns confluence user link
+   */
+  const getNewUserArray = () =>
+    getStringsFromInput()
+      .map(
+        (text) =>
+          '<a class="confluence-userlink user-mention current-user-mention userlink-0" data-username="' +
+          text +
+          '" href="/confluence/display/~' +
+          text +
+          '" data-linked-resource-id="65588" data-linked-resource-version="1" data-linked-resource-type="userinfo" ' +
+          'data-base-url="' +
+          AJS.Confluence.getBaseUrl() +
+          ' title="" data-user-hover-bound="true">' +
+          text +
+          "</a>"
+      )
+      .join(", ");
+
+  // The category assigned to the macro
+  const categoryArray = macroNode.dataset.macroParameters.split("=");
+  // Put the macro params into the correct format
+  const macroParams = { [categoryArray[0]]: categoryArray[1] };
+
+  const handleFormSubmit = (category, newBodyText) => {
+    const formID = "#edit-page-dialog-" + category;
```

Comment:
A template literal can be used here

```suggestion
    const formID = `#edit-page-dialog-${category}`;
```

Reply:
Updated.